numba/cuda/cudadecl.py

```diff
@@ -529,6 +565,51 @@ def resolve_habs(self, mod):
     def resolve_hfma(self, mod):
         return types.Function(Cuda_hfma)
 
+    def resolve_hsin(self, mod):
+        return hsin_device
```

Comment:
Example use of `resolve_wrapped_unary`:

```suggestion
        return self._resolve_wrapped_unary('hsin')
```